src/foam/nanos/approval/ApprovalRequestUtil.java

```diff
@@ -77,6 +77,7 @@ public void put(Object obj, Detachable sub) {
     } catch (Exception e){
       return tester.getState();
     }
-    return tester.getState();
+    ApprovalStatus state = tester.getState();
+    return state == null ? ApprovalStatus.REQUESTED : state;
```

Comment:
if the collection is empty it would return REQUESTED status, is it intended?

Reply:
good point, probably not .. will change this, thanks